packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -288,6 +300,80 @@ class FuchsiaDevice extends Device {
     throwToolExit('No ports found running $isolateName');
     return null;
   }
+
+  FuchsiaIsolateDiscoveryProtocol  getIsolateDiscoveryProtocol(String isolateName) => FuchsiaIsolateDiscoveryProtocol(this, isolateName);
+}
+
+class FuchsiaIsolateDiscoveryProtocol {
```

Comment:
There's some non-trivial logic in here. Are the tests of 'attach' enough to cover the new polling code?

Reply:
No, added some test cases